### PR TITLE
5.x.OIDC fix post endsession and authorize

### PR DESCRIPTION
**Pull request by gianpaolo-tndigit**
add ignore path for CSRF for:
"/endsession"
"/oauth/authorize"

to permit POST request as specifics OIDC

### CODE REVIEW

File: src/main/java/it/smartcommunitylab/aac/config/SecurityConfig.java [hunk 1]
```diff
@@ -178,7 +179,7 @@ public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
             //                .and()
             .csrf()
             //                .disable()
-            .ignoringAntMatchers("/logout", "/console/**", "/account/**")
+            .ignoringAntMatchers("/logout", "/console/**", "/account/**", EndSessionEndpoint.END_SESSION_URL)
```
Comment (matteo-s) [inline src/main/java/it/smartcommunitylab/aac/config/SecurityConfig.java:182]: remove endsession from this filter and move to OAuth2UserSecurityConfig, both for csrf whitelist and permitAll

**Review by matteo-s — CHANGES_REQUESTED** [review]

File: src/main/java/it/smartcommunitylab/aac/config/SecurityConfig.java [hunk 2]
```diff
@@ -21,6 +21,7 @@
 import it.smartcommunitylab.aac.core.entrypoint.RealmAwarePathUriBuilder;
 import it.smartcommunitylab.aac.core.provider.ProviderConfigRepository;
 import it.smartcommunitylab.aac.crypto.InternalPasswordEncoder;
+import it.smartcommunitylab.aac.openid.endpoint.EndSessionEndpoint;
```
Comment (matteo-s) [inline src/main/java/it/smartcommunitylab/aac/config/SecurityConfig.java:24]: remove the unused import

Comment (gianpaolo-tndigit) [reply src/main/java/it/smartcommunitylab/aac/config/SecurityConfig.java]: solved with 8136bdcd8efb357e56235b44e1fd6b3186c0582a

File: src/main/java/it/smartcommunitylab/aac/config/OAuth2UserSecurityConfig.java [hunk 3]
```diff
@@ -106,8 +107,10 @@ public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
         // match only user endpoints
         http
             .requestMatcher(getRequestMatcher())
-            //                .authorizeRequests().anyRequest().authenticated().and()
-            .authorizeRequests(authorizeRequests -> authorizeRequests.anyRequest().hasAnyAuthority(Config.R_USER))
+            //.authorizeRequests().anyRequest().authenticated().and()
```
[the remark this answers is not in the document]
Comment (gianpaolo-tndigit) [reply src/main/java/it/smartcommunitylab/aac/config/OAuth2UserSecurityConfig.java:110]: resolve with 17c2c28b727e475cba6957d354fbad20d690a9cc

**Pull request merged**